### ISSUE TITLE
file_header rule should trigger on first line if there're no comments

### ISSUE THREAD

**Issue by rjhodge** [realm/SwiftLint#1520]
So I have the file header rule working great with one small problem. One developer is notorious for not having a file header. How can I enforce requiring this?

**Comment by marcelofabri** [comment]
Do you want to enforce a certain header format? Take a look how SwiftLint itself does it: https://github.com/realm/SwiftLint/blob/master/.swiftlint.yml#L25-L33

**Comment by rjhodge** [comment]
So I have the header format

```
file_header:
  required_pattern: |
                    \/\/
                    \/\/  .*?\.swift
                    \/\/  Project
                    \/\/
                    \/\/  Created by .*? on \d{1,2}\/\d{1,2}\/\d{2}\.
                    \/\/  Copyright © \d{4} Company\. All rights reserved\.
                    \/\/
```

But he will often put in the following for files

```
import MapKit
import UIKit
```

And this still passes the file_header rule.

**Comment by marcelofabri** [comment]
This rule is opt-in, are you sure you've enabled it as well? It should trigger a warning.

**Comment by rjhodge** [comment]
We are using the whitelist_rules and have it listed as one of the rules. And the rule works great at finding inconsistent headers but does nothing when there is no header.

**Comment by marcelofabri** [comment]
Can you provide a small example that triggers the issue? I couldn't reproduce it:

```
$ swiftlint      
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'file.swift' (1/1)
/Users/marcelofabri/Documents/file.swift: warning: File Header Violation: Files should have consistent header comments. (file_header)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**`file.swift`**
```swift
import MapKit
import UIKit
```

**`.swiftlint.yml`**
```yml
opt_in_rules:
  - file_header

file_header:
  required_pattern: |
                    \/\/
                    \/\/  .*?\.swift
                    \/\/  Project
                    \/\/
                    \/\/  Created by .*? on \d{1,2}\/\d{1,2}\/\d{2}\.
                    \/\/  Copyright © \d{4} Company\. All rights reserved\.
                    \/\/
```

**Comment by marcelofabri** [comment]
Maybe the issue is that when there's a comment, the warning is triggered on that line. Otherwise, the warning is only on the file. That could make Xcode behaves differently.

**Comment by rjhodge** [comment]
Ahhh that appears to be the issue. Xcode is ditching the warning. Running it from terminal the warning shows up properly.

Anyway to fix it for Xcode?

**Comment by marcelofabri** [comment]
Not that I'm aware of. But it'd be possible to change the violation to trigger on the first line of the file here: https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/FileHeaderRule.swift#L86.

Do you want to try that? Otherwise I might be able to take a look later today.

**Comment by rjhodge** [comment]
I'll give it a shot.

**Comment by marcelofabri** [comment]
BTW, I just tried on Xcode and I see the warning on the issue navigator, but not on the file editor itself.

**Comment by rjhodge** [comment]
Ok this is fixed in the pull request #1521 please review when you get a chance.

Thanks again for the help.
